Mathlib/Algebra/Order/CauSeq/Completion.lean

```diff
@@ -185,6 +183,7 @@ instance instRatCast : RatCast (Cauchy abv) where ratCast q := ofRat q
 @[simp, norm_cast] lemma ofRat_nnratCast (q : ℚ≥0) : ofRat (q : β) = (q : Cauchy abv) := rfl
 @[simp, norm_cast] lemma ofRat_ratCast (q : ℚ) : ofRat (q : β) = (q : Cauchy abv) := rfl
 
+open Classical in
```

Comment:
Same question here.

Reply:
Here it's an instance, so I definitely don't want decidability assumptions, to make sure it's available all the time. In general, in definitions and instances, `open Classical in` is always the way to go, while in theorems one has to double-think.

Reply:
We also have instancen which explicitly take a `DecidableXX` assumption...